DBAgenda.php

```diff
@@ -0,0 +1,311 @@
+<?php
+
+use Monolog\Logger;
+use Sabre\VObject;
+
+abstract class DBAgenda extends Agenda {
+    protected $pdo;
+    
+    public function __construct(string $url, string $username, string $password, object $api, array $agenda_args) {
```

Comment:
Nitpicking: the `$username` and `$password` parameter could be renamed to something like `$caldavUsername` and `$caldavPassword` (because in the context of a class with `DB` in the name, it could be interpreted as the credentials of the db)